src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -14,36 +14,105 @@ namespace Microsoft.Health.Events.Telemetry.Exceptions
+                            message = "Verify that the provided Event Hub's consumer group is not already receiving data for another IoT connector or Azure resource.";
+                            break;
+                        case EventHubsException.FailureReason.ResourceNotFound:
+                            message = "Verify that the provided Event Hubs Namespace contains the provided Event Hub and that the provided Event Hub contains the provided consumer group.";
+                            break;
+                        case EventHubsException.FailureReason.ServiceCommunicationProblem:
+                            message = "Verify that the provided Event Hub Namespace, Event Hub name, and consumer group are correct and that access permissions to the provided Event Hub have been granted.";
+                            break;
+                        default:
+                            return exception;
+                    }
+
+                    return new InvalidEventHubException(message, exception, nameof(EventHubErrorCode.ConfigurationError));
+
+                case InvalidOperationException _:
+                    message = "Verify that the provided Event Hub contains the provided consumer group.";
+                    return new InvalidEventHubException(message, exception, nameof(EventHubErrorCode.ConfigurationError));
+
                 case SocketException _:
-                    return $"{ErrorType.EventHubError}{EventHubErrorCode.SocketError}";
+                    SocketException socketException = (SocketException)exception;
+                    switch (socketException.SocketErrorCode)
+                    {
+                        case SocketError.HostNotFound:
+                            message = "Verify that the provided Event Hubs Namespace exists.";
+                            return new InvalidEventHubException(message, exception, nameof(EventHubErrorCode.ConfigurationError));
+                        default:
+                            return exception;
+                    }
+
                 case UnauthorizedAccessException _:
-                    return $"{ErrorType.EventHubError}{EventHubErrorCode.AuthorizationError}";
+                    message = "Verify that the provided Event Hub's 'Azure Event Hubs Data Receiver' role has been assigned to the applicable Azure Active Directory security principal or managed identity.";
+                    string helpLink = "https://docs.microsoft.com/azure/event-hubs/authenticate-application";
+                    return new UnauthorizedAccessEventHubException(message, exception, helpLink, nameof(EventHubErrorCode.AuthorizationError));
+
                 default:
-                    return $"{ErrorType.EventHubError}{EventHubErrorCode.GeneralError}";
+                    return exception;
             }
         }
+
+        private static string GetErrorMetricName(Exception exception)
```

Comment:
As discussed, can avoid recasting the exceptions and have the previous one also return the errorname additionally.